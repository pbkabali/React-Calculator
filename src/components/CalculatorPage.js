import Calculator from './Calculator';

const CalculatorPage = () => <Calculator />;

export default CalculatorPage;
