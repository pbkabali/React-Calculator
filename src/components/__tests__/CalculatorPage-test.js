import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import CalculatorPage from '../CalculatorPage';

describe('CalculatorPage', () => {
  it('Renders correctly', () => {
    const page = renderer.create(<CalculatorPage />);
    expect(page).toMatchSnapshot();
  });

  it('Caculator displays a zero upon launch', async () => {
    render(<CalculatorPage />);
    const res = await waitFor(() => screen.findAllByText('0'));
    expect(res).toHaveLength(2);
  });

  it('Displays clicked number', async () => {
    render(<CalculatorPage />);
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('5'));
    const res = await waitFor(() => screen.findAllByText('95'));
    expect(res).toHaveLength(1);
  });

  it('Displays sum on adding two numbers', async () => {
    render(<CalculatorPage />);
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));
    const res = await waitFor(() => screen.findAllByText('14'));
    expect(res).toHaveLength(1);
  });

  it('Displays product on multiplying two numbers', async () => {
    render(<CalculatorPage />);
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));
    const res = await waitFor(() => screen.findAllByText('45'));
    expect(res).toHaveLength(1);
  });

  it('Changes the sign of the displayed figure when "+/-" is clicked', async () => {
    render(<CalculatorPage />);
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+/-'));
    const res = await waitFor(() => screen.findAllByText('-95'));
    expect(res).toHaveLength(1);
  });

  it('Displays the percentantage of figure when "%" is clicked', async () => {
    render(<CalculatorPage />);
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('%'));
    const res = await waitFor(() => screen.findAllByText('0.95'));
    expect(res).toHaveLength(1);
  });
});
