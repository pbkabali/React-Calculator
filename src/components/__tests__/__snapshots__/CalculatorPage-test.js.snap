// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorPage Renders correctly 1`] = `
<div
  className="flex side-padding calc-container"
>
  <h2>
    Let's do some math!
  </h2>
  <div>
    <div
      className="flex calculator display"
    >
      0
    </div>
    <div>
      <div
        className="flex calculator"
      >
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          /
        </button>
      </div>
      <div
        className="flex calculator"
      >
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="flex calculator"
      >
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="flex calculator"
      >
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="flex calculator"
      >
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "50%",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="btn-normal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
